### PR TITLE
chore: Improve the iOS CI job

### PR DESCRIPTION
**Pull request by jbelkins**
## Description of changes
Replaces the iOS Github CI job with a new job having the following improvements:
- Does not use the CRT Builder script.  Provides its own lightweight scripting for project setup.
- No dependency on access to any AWS account or any special Github credentials.
- No need to maintain any external build container or other resource.
- Builds protocol tests & unit tests together, reducing build time by de-duplicating several build steps.
- Uses `xcodebuild` to build & test the SDK using the iOS Simulator.
- `xcbeautify` is used to clean up & reduce the volume of Swift logs, and to reduce build time.

Once this job has had an opportunity to run for a while and proves reliable & efficient, I plan to adapt it for use with other CI jobs as well.

## New/existing dependencies impact assessment, if applicable
No new dependencies were added to this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -24,33 +24,65 @@ jobs:
...
+          restore-keys: |
+            1-${{ runner.os }}-swift5.8.1-spm-${{ hashFiles('Package.swift', 'AWSSDKSwiftCLI/Package.swift') }}
+            1-${{ runner.os }}-swift5.8.1-spm-
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: corretto
+          java-version: 17
+      - name: Setup xcbeautify
+        run: HOMEBREW_NO_AUTO_UPDATE=1 brew install xcbeautify
```
Comment (waahm7) [inline .github/workflows/continuous-integration.yml:70]: Why set `HOMEBREW_NO_AUTO_UPDATE`? Each CI run will be re-installing xcbeautify.

**Review by waahm7 — APPROVED** [review]

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (jbelkins) [reply .github/workflows/continuous-integration.yml:70]: Homebrew will automatically update whenever it's invoked, unless this var is set.

The Github CI machines are often pretty far behind on the Homebrew repo, so updating homebrew takes a fair amount of time.

`xcbeautify` itself doesn't get updated very often, and I'm OK with being behind the latest version.

**Pull request merged**